src/sets.jl

```diff
@@ -240,3 +239,227 @@ function distance_to_set(d::DefaultDistance, v::AbstractVector{T}, s::MOI.Indica
         (distance_to_set(d, z, MOI.ZeroOne()), distance_to_set(v[2], s.set))
     )
 end
+
+
+# find expression of projections on cones and their derivatives here:
+#   https://stanford.edu/~boyd/papers/pdf/cone_prog_refine.pdf
+
+
+"""
+    projection of vector `z` on zero cone i.e. K = {0} or its dual
+"""
+function projection_on_set(::DefaultDistance, ::MOI.Zeros, z::Array{Float64}, dual=true)
+    return dual ? z : zeros(Float64, size(z))
+end
+
+function projection_on_set(::DefaultDistance, ::MOI.EqualTo, z::Array{Float64}, dual=true)
+    return dual ? z : zeros(Float64, size(z))
+end
+
+function projection_on_set(::DefaultDistance, ::MOI.EqualTo, z::Float64, dual=true)
```

Comment:
I don't think this is correct.
Projection on s::MOI.EqualTo
is always equal to the value of the set

Reply:
yeah thanks for pointing out the bug

Reply:
in the test examples, i had an equal to constraint w/ value 0, thats why it did'nt break

Reply:
I see, add other examples with random EqualTo